File: Java/src/main/java/com/gildedrose/SulfurasItem.java
package com.gildedrose;

public class SulfurasItem extends BaseItem {

    protected SulfurasItem(String name, int sellIn, int quality) {
        super(name, sellIn, quality);
    }

    @Override
    protected void updateQuality() {
        // never change quality
    }

    @Override
    protected void updateSellIn() {
        // never change sellin
    }

}

File: scala/src/main/scala/com/gildedrose/SulfurasItem.scala
package com.gildedrose

class SulfurasItem(item: Item) extends BaseItem(item) {

  override protected def updateSellIn: Unit = {
    // don't update sellin
  }

  override protected def updateQuality: Unit = {
    // don't update quality
  }

}

object SulfurasItem extends BaseFactory {

  def create(item: Item): Option[BaseItem] = if ("Sulfuras, Hand of Ragnaros".equals(item.name)) Option(new SulfurasItem(item)) else None;

}
